### PR TITLE
Add a new general function that checks the sum spectrum's state 

### PR DESCRIPTION
**Pull request by jkpnen**
FIX: #218

### CODE REVIEW

File: modules/general_functions.py [hunk 1]
```diff
@@ -656,3 +656,25 @@ def find_next(iterable: Iterable[T], cond: Callable[[T], bool]) -> T:
         return next(i for i in iterable if cond(i))
     except StopIteration:
         raise ValueError("Value not found in iterable.")
+
+
+def check_if_sum_spectrum_is_selected(directory):
+    """
+    Check if the sum spectrum is selected on the GUI.
```
Comment (tpitkanen) [inline modules/general_functions.py:663]: The docstring doesn't match what the function does at all. There is no GUI involved here.

File: modules/general_functions.py [hunk 2]
```diff
@@ -656,3 +656,25 @@ def find_next(iterable: Iterable[T], cond: Callable[[T], bool]) -> T:
...
+
+    """
+    measured_sum = False
+    simulated_sum = False
+    for filepath in os.listdir(directory):
+        if "MEASURED_SUM" in filepath:
+            measured_sum = True
+        if "SIMULATED_SUM" in filepath:
+            simulated_sum = True
+    return measured_sum, simulated_sum
```
Comment (tpitkanen) [inline modules/general_functions.py:680]:
```suggestion
    measured_sum_found = False
    simulated_sum_found = False
    with os.scandir(directory) as sdir:
        for entry in sdir:
            if entry.name.startswith("MEASURED_SUM"):
                measured_sum_found = True
            if entry.name.startswith("SIMULATED_SUM"):
                simulated_sum_found = True
    return measured_sum_found, simulated_sum_found

```

Improvements:
- checking for "MEASURED_SUM"/"SIMULATED_SUM" anywhere in the path could lead to false positives, entry.name is less likely to do so
- `os.scandir` is faster and always returns a path-like object, unlike `os.listdir` which is more complicated
- more descriptive variable names

Also, those "MEASURED_SUM" and "SIMULATED_SUM" strings should be constants that are defined in one place (not inside the function).

**Review by tpitkanen — CHANGES_REQUESTED** [review]

File: modules/general_functions.py (hunk 1 above)
Comment (tpitkanen) [reply modules/general_functions.py:663]: Same goes for the function name, a more exact name would be something like `check_for_sum_spectra_files(directory)`

Comment (jkpnen) [reply modules/general_functions.py:663]: Pedantry. It does not **literally** check it from the GUI but it does it implicitly. However, I'll change that.

File: modules/general_functions.py (hunk 2 above)
Comment (jkpnen) [reply modules/general_functions.py]: OK

**Review by tpitkanen — APPROVED** [review]

**Pull request merged**